vm/src/function.rs

```diff
@@ -46,6 +47,12 @@ impl From<(&Args, &KwArgs)> for PyFuncArgs {
     }
 }
 
+impl FromArgs for PyFuncArgs {
```

Comment:
Not really a fan of this impl because it's not obvious that it will force a pointless allocation of a `PyFuncArgs`. And there's nothing stopping someone from trying to combine this with other extractors.

Reply:
`Vec::new()` doesn't allocate, so replacing it with an empty `PyFuncArgs` effectively zeros's the struct. Yes there is nothing to stop people from misusing this, but this is also true for the existing extractors like `Args`, `KwArgs`, and `OptionalArg`.

Reply:
Fair enough, still seems a bit weird to me to have `PyFuncArgs` implement `FromArgs`, but it's not a huge deal